test/integration/src/test/java/io/pravega/test/integration/LargeEventTest.java

```diff
@@ -0,0 +1,510 @@
+                .scalingPolicy(ScalingPolicy.byDataRate(500, 2, 1))
+                .retentionPolicy(RetentionPolicy.bySizeBytes(Long.MAX_VALUE))
+                .build();
+
+        createScopeStream(SCOPE_NAME, streamName, config);
+        generateWriteEventData(NUM_WRITERS, LARGE_EVENT_SIZE);
+
+        try (ConnectionExporter connectionFactory = new ConnectionExporter(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE_NAME, controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(SCOPE_NAME, controller, clientFactory)) {
+            // Start writing events to the stream.
+            val writers = createEventWriters(streamName, NUM_WRITERS, clientFactory);
+            // Create a ReaderGroup.
+            createReaderGroup(readerGroupName, readerGroupManager, streamName);
+            // Create Readers.
+            val readers = createEventReaders(NUM_READERS, clientFactory, readerGroupName);
+            Futures.allOf(writers).get();
+            stopReadFlag.set(true);
+
+            Futures.allOf(readers).get();
+
+            log.info("Deleting ReaderGroup: {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+
+        validateCleanUp(streamName);
+        validateEventReads(eventsReadFromPravega, eventsWrittenToPravega);
+    }
+
+    private CompletableFuture<Void> startNewWriter(final String routingKey,
+                                                   final String streamName,
+                                                   final AtomicLong writeCount,
+                                                   final ByteBuffer data,
+                                                   final EventStreamClientFactory clientFactory) {
+        return startNewWriterPreflushAction(routingKey, streamName, writeCount, data, clientFactory, null);
+    }
+
+    private CompletableFuture<Void> startNewWriterPreflushAction(final String routingKey,
+                                                   final String streamName,
+                                                   final AtomicLong writeCount,
+                                                   final ByteBuffer data,
+                                                   final EventStreamClientFactory clientFactory,
+                                                         Runnable action) {
+        return CompletableFuture.runAsync(() -> {
+            @Cleanup
+            final EventStreamWriter<ByteBuffer> writer = clientFactory.createEventWriter(streamName,
+                    new ByteBufferSerializer(),
+                    EventWriterConfig.builder().build());
+
+            log.debug("Writing Large Event : {}", routingKey);
+            writeCount.incrementAndGet();
+            writer.writeEvent(routingKey, data);
+            // Allow one to supply an action during a write call, such as a SegmentStore restart.
```

Comment:
As written this is not 'during' the write call. It's afterwards, which isn't an interesting validation for a segment store restart.